2024-03-01 10:38:46 error: uncaughtException: Route.get() requires a callback function but got a [object Undefined]
Error: Route.get() requires a callback function but got a [object Undefined]
    at Route.<computed> [as get] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:211:15)
    at proto.<computed> [as get] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:521:19)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\article.js:16:8)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\index.js:3:23)
2024-03-01 10:38:55 error: uncaughtException: Route.get() requires a callback function but got a [object Undefined]
Error: Route.get() requires a callback function but got a [object Undefined]
    at Route.<computed> [as get] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:211:15)
    at proto.<computed> [as get] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:521:19)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\article.js:16:8)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\index.js:3:23)
2024-03-01 10:39:01 error: uncaughtException: Route.get() requires a callback function but got a [object Undefined]
Error: Route.get() requires a callback function but got a [object Undefined]
    at Route.<computed> [as get] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:211:15)
    at proto.<computed> [as get] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:521:19)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\article.js:26:8)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\index.js:3:23)
2024-03-01 11:00:23 error: Error: Authentication Failed,    at authenticationError (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:9:17),    at exports.isAuthenticated (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\authMiddleware.js:36:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at Route.dispatch (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:114:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:284:15,    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:365:14),    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:376:14),    at Function.process_params (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:421:3)
2024-03-01 11:00:27 error: Error: Authentication Failed,    at authenticationError (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:9:17),    at exports.isAuthenticated (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\authMiddleware.js:36:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at Route.dispatch (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:114:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:284:15,    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:365:14),    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:376:14),    at Function.process_params (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:421:3)
2024-03-01 11:01:31 error: Error: Authentication Failed,    at authenticationError (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:9:17),    at exports.isAuthenticated (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\authMiddleware.js:36:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at Route.dispatch (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:114:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:284:15,    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:365:14),    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:376:14),    at Function.process_params (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:421:3)
2024-03-01 11:01:34 error: Error: Authentication Failed,    at authenticationError (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:9:17),    at exports.isAuthenticated (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\authMiddleware.js:36:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at Route.dispatch (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:114:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:284:15,    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:365:14),    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:376:14),    at Function.process_params (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:421:3)
2024-03-01 11:03:22 error: Error: Authentication Failed,    at authenticationError (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:9:17),    at exports.isAuthenticated (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\authMiddleware.js:36:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at Route.dispatch (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:114:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:284:15,    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:365:14),    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:376:14),    at Function.process_params (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:421:3)
2024-03-01 11:05:12 error: Error: Authentication Failed,    at authenticationError (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:9:17),    at exports.isAuthenticated (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\authMiddleware.js:36:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at Route.dispatch (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:114:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:284:15,    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:365:14),    at param (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:376:14),    at Function.process_params (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:421:3)
2024-03-01 11:07:53 error: Error: Resource not found with this 65e09d9254a397fa153f8321,    at notFound (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:15:17),    at ArticleService.deleteById (G:\Programming\projects\FullStack\MERN Blog\server\src\service\ArticleService.js:51:13),    at process.processTicksAndRejections (node:internal/process/task_queues:95:5),    at async G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\article.js:56:17
2024-03-01 11:08:11 error: Error: Bad Request,    at badRequest (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:2:17),    at requestValidation (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\requestValidationMiddleware.js:10:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at middleware (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express-validator\src\middlewares\check.js:16:13),    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-03-01 22:42:33 error: Error: Bad Request,    at badRequest (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:2:17),    at requestValidation (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\requestValidationMiddleware.js:10:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at middleware (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express-validator\src\middlewares\check.js:16:13),    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-03-01 22:42:55 error: Error: Bad Request,    at badRequest (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:2:17),    at requestValidation (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\requestValidationMiddleware.js:10:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at middleware (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express-validator\src\middlewares\check.js:16:13),    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-03-01 22:51:31 error: Error: Authentication Failed,    at authenticationError (G:\Programming\projects\FullStack\MERN Blog\server\src\utils\error.js:9:17),    at exports.isAuthenticated (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\authMiddleware.js:36:17),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at Route.dispatch (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:114:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:284:15,    at Function.process_params (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:346:12),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:280:10),    at Function.handle (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:175:3)
2024-03-01 22:52:27 error: TypeError: Cannot read properties of undefined (reading 'path'),    at AuthService.signup (G:\Programming\projects\FullStack\MERN Blog\server\src\service\AuthService.js:34:64),    at G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\auth.js:24:33,    at G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\catchAsyncErrors.js:7:19,    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at requestValidation (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\requestValidationMiddleware.js:13:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at middleware (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express-validator\src\middlewares\check.js:16:13),    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-03-01 22:57:09 error: TypeError: Cannot read properties of undefined (reading 'path'),    at AuthService.signup (G:\Programming\projects\FullStack\MERN Blog\server\src\service\AuthService.js:34:64),    at G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\auth.js:24:33,    at G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\catchAsyncErrors.js:7:19,    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at requestValidation (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\requestValidationMiddleware.js:13:3),    at Layer.handle [as handle_request] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\layer.js:95:5),    at next (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:144:13),    at middleware (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express-validator\src\middlewares\check.js:16:13),    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-03-01 22:59:49 error: TypeError: Cannot read properties of null (reading 'public_id'),    at AuthService.signup (G:\Programming\projects\FullStack\MERN Blog\server\src\service\AuthService.js:47:33),    at async G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\auth.js:24:15
2024-03-01 23:24:06 error: uncaughtException: Identifier 'user' has already been declared
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:5
    let user = await UserRepository.findByID(user._id, {
        ^

SyntaxError: Identifier 'user' has already been declared
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:24:16 error: uncaughtException: Identifier 'user' has already been declared
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:5
    let user = await UserRepository.findByID(user._id, {
        ^

SyntaxError: Identifier 'user' has already been declared
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:24:38 error: uncaughtException: Identifier 'user' has already been declared
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:5
    let user = await UserRepository.findByID(user._id, {
        ^

SyntaxError: Identifier 'user' has already been declared
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:24:50 error: uncaughtException: Identifier 'user' has already been declared
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:5
    let user = await UserRepository.findByID(user._id, {
        ^

SyntaxError: Identifier 'user' has already been declared
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:24:54 error: uncaughtException: Identifier 'user' has already been declared
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:5
    let user = await UserRepository.findByID(user._id, {
        ^

SyntaxError: Identifier 'user' has already been declared
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:25:06 error: uncaughtException: Identifier 'user' has already been declared
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:5
    let user = await UserRepository.findByID(user._id, {
        ^

SyntaxError: Identifier 'user' has already been declared
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:25:18 error: uncaughtException: Identifier 'user' has already been declared
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:5
    let user = await UserRepository.findByID(user._id, {
        ^

SyntaxError: Identifier 'user' has already been declared
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:25:25 error: uncaughtException: Identifier 'user' has already been declared
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:5
    let user = await UserRepository.findByID(user._id, {
        ^

SyntaxError: Identifier 'user' has already been declared
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:27:55 error: uncaughtException: Route.get() requires a callback function but got a [object Undefined]
Error: Route.get() requires a callback function but got a [object Undefined]
    at Route.<computed> [as get] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\route.js:211:15)
    at proto.<computed> [as get] (G:\Programming\projects\FullStack\MERN Blog\server\node_modules\express\lib\router\index.js:521:19)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\user.js:13:8)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\index.js:2:20)
2024-03-01 23:35:59 error: uncaughtException: await is only valid in async functions and the top level bodies of modules
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:14
    let profile = await UserRepository.findByID(user._id, {
                  ^^^^^

SyntaxError: await is only valid in async functions and the top level bodies of modules
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:36:11 error: uncaughtException: await is only valid in async functions and the top level bodies of modules
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:14
    let profile = await UserRepository.findByID(username, {
                  ^^^^^

SyntaxError: await is only valid in async functions and the top level bodies of modules
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:36:17 error: uncaughtException: await is only valid in async functions and the top level bodies of modules
G:\Programming\projects\FullStack\MERN Blog\server\src\service\UserService.js:14
    let profile = await UserRepository.findByUsername(username, {
                  ^^^^^

SyntaxError: await is only valid in async functions and the top level bodies of modules
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\controllers\user.js:4:21)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:39:17 error: uncaughtException: Unexpected identifier 'findByUsernameWithArticles'
G:\Programming\projects\FullStack\MERN Blog\server\src\repository\userRepository.js:23
  const findByUsernameWithArticles = (username, select = {}) => {
        ^^^^^^^^^^^^^^^^^^^^^^^^^^

SyntaxError: Unexpected identifier 'findByUsernameWithArticles'
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\middlewarers\authMiddleware.js:2:24)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
2024-03-01 23:50:20 error: uncaughtException: getUserProfile is not defined
ReferenceError: getUserProfile is not defined
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\user.js:7:28)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (G:\Programming\projects\FullStack\MERN Blog\server\src\routes\index.js:2:20)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
